File: Data Structures & Algorithms/is-anagram/submission-7.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
       vector<int>set(26,0);
       int n= s.size();
       for(int i=0; i<n;i++){
            set[s[i]-'a']++;
            set[t[i]-'a']--;
       }
       for(auto val:set){
        if(val!=0){
            return false;
        }
       }
       return true;
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-7.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int n=nums.length;
        int l=0,r=0;
        int[ ] ans=new int [n-k+1];
        Deque<Integer> dq= new LinkedList<>();
        while(r<n){
            while(!dq.isEmpty() && nums[dq.peekLast()]<nums[r]){
                dq.removeLast();
            }
            dq.offerLast(r);
            if(!dq.isEmpty() && dq.peekFirst()<l){
                dq.removeFirst();
            }
            
            if(r+1>=k){
                ans[l]=nums[dq.peekFirst()];
                l++;
            }
            r++;
        }
        return ans;
    }
}
